### PR TITLE
feat: firefox devedition archive versions

### PR DESCRIPTION
**Pull request by DanielSinclair**
## Changes
- added ability to accept product versions in the `version` param for archive download URLs; this allows downloading historical devedition releases
- product name parsing if included in `version` (i.e. `devedition-80.0b1` or `firefox-80.0`)
- version parsing to ignore product name if specified in download URL generation
- updated readme to highlight `devedition` versioning
- added `firefox-developer` CLI test in build GitHub Action

## Tests
- coverage of product-specified versions in `ArchiveDownloadURL`, including `firefox-80.0`, `devedition-80.0b1`, and `beta-80.0b1`
- coverage of product-specified versions in `DownloadURLFactory`, including `firefox-80.0`, `beta-80.0b1`, and `devedition-80.0b1`

### CODE REVIEW

**Comment by DanielSinclair** [comment]
@ueokande Our team noticed that we were unable to install static/historical versions of `devedition` after the `latest-devedition` introduced an issue, so I've added that ability in this PR

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -50,3 +50,8 @@ jobs:
           firefox-version: "84.0"
       - run: |
           firefox --version
+      - uses: ./
+        with:
+          firefox-version: "devedition-84.b1"
```
Comment (ueokande) [inline .github/workflows/build.yml]: The test [fails](https://github.com/browser-actions/setup-firefox/actions/runs/7716884084/job/21678745839#step:9:8).  I think it should be `devedition-84.0b1`

```suggestion
          firefox-version: "devedition-84.0b1"
```

**Review by ueokande — CHANGES_REQUESTED** [review]

**Comment by DanielSinclair** [comment]
@ueokande Thanks! Should be fixed now

**Review by ueokande — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**